devlib/module/cgroups.py

```diff
@@ -287,6 +287,11 @@ def get_tasks(self):
         logging.debug('Tasks: %s', task_ids)
         return map(int, task_ids)
 
+    # Used to insert fake cgroup attach events to know existing cgroup assignments
+    def trace_cgroup_tasks(self):
```

Comment:
Moreover, I think we should also add a method to the `CgroupsModule` class, which allows to trace the cgroups for all the CGroups of all the currently mounted controllers.
I think this will be a quite common use case and we don't want to have this code in the clients of the CGroups API. Possibly, this new call should also allow to specify a "mask" of cgroups names to report. Something like:

```python
class CgroupModule(Module):

    def trace_cgroups(self, controllers=None, cgroups=None):
        if controllers is None:
            controllers = [ss.name for ss in self.controllers]
        for controller in self.controllers:
            if controller not in controllers:
                continue
            controller.trace_cgroups(cgroups)
```

Which ultimately requires an additional method in the `Contoller` class doing almost the same but filtering on cgroups names to call your method above just on listed cgroups (or all if None).

If you don't have time, I can provide such a patch in the next few days...